internal/pkg/provision/providers/vm/dhcpd.go

```diff
@@ -58,6 +58,26 @@ func handler(serverIP net.IP, statePath string) server4.Handler {
 			return
 		}
 
+		if m.IsOptionRequested(dhcpv4.OptionBootfileName) {
+			log.Printf("received PXE boot request from %s", m.ClientHWAddr)
+
+			if match.TFTPServer != "" {
+				bootFilename := match.BootFilename
+
+				for _, userClass := range m.UserClass() {
```

Comment:
looks like this is always true for QEMU, so we can simplify this to use single IPXEBootFilename